sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -537,13 +540,27 @@ public void addIntegration(@NotNull Integration integration) {
     return dsn;
   }
 
+  /**
+   * Returns the DSN
+   *
+   * @return the DSN or null if not set
+   */
+  @ApiStatus.Internal
+  @NotNull
+  Dsn getParsedDsn() {
+    return parsedDsn.getValue();
+  }
+
   /**
    * Sets the DSN
    *
    * @param dsn the DSN
    */
   public void setDsn(final @Nullable String dsn) {
     this.dsn = dsn;
+    if (!isEnabled() || (dsn != null && dsn.isEmpty())) {
+      this.parsedDsn.resetValue();
+    }
```

Comment:
I would always reset the value, ensuring it's in sync with `dsn`
```suggestion
this.parsedDsn.resetValue();
```